api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -651,20 +651,92 @@ private void mergeVisits(Patient preferred, Patient notPreferred, PersonMergeLog
 		//TODO: this should be a copy, not a move
 		
 		VisitService visitService = Context.getVisitService();
-		
-		for (Visit visit : visitService.getVisitsByPatient(notPreferred, true, true)) {
-			if (log.isDebugEnabled()) {
-				log.debug("Merging visit " + visit.getVisitId() + " to " + preferred.getPatientId());
-			}
-			visit.setPatient(preferred);
-			Visit persisted = visitService.saveVisit(visit);
-			mergedData.addMovedVisit(persisted.getUuid());
-		}
-	}
+		EncounterService encounterService = Context.getEncounterService();
+
+		List<Visit> preferredVisits = visitService.getVisitsByPatient(preferred, true, true);
+		List<Visit> notPreferredVisits = visitService.getVisitsByPatient(notPreferred, true, true);
+		
+		for (Visit losing : notPreferredVisits) {   
+			if (!losing.isVoided()) {      
```

Comment:
Did you take a look at here,
`getVisitsByPatient(Patient patient, boolean includeInactive, boolean includeVoided)`  
You can only fetch the notPreferredVisits list without voided visits.